pkg/cluster/k8sres.go

```diff
@@ -840,21 +847,71 @@ func (c *Cluster) generateStatefulSet(spec *acidv1.PostgresSpec) (*appsv1.Statef
 	}
 
 	// generate environment variables for the spilo container
-	spiloEnvVars := deduplicateEnvVars(
-		c.generateSpiloPodEnvVars(c.Postgresql.GetUID(), spiloConfiguration, &spec.Clone,
-			spec.StandbyCluster, customPodEnvVarsList), c.containerName(), c.logger)
+	spiloEnvVars := c.generateSpiloPodEnvVars(
+		c.Postgresql.GetUID(),
+		spiloConfiguration,
+		&spec.Clone,
+		spec.StandbyCluster,
+		customPodEnvVarsList,
+	)
 
 	// pickup the docker image for the spilo container
 	effectiveDockerImage := util.Coalesce(spec.DockerImage, c.OpConfig.DockerImage)
 
+	// determine the FSGroup for the spilo pod
+	effectiveFSGroup := c.OpConfig.Resources.SpiloFSGroup
+	if spec.SpiloFSGroup != nil {
+		effectiveFSGroup = spec.SpiloFSGroup
+	}
+
 	volumeMounts := generateVolumeMounts(spec.Volume)
 
+	// configure TLS with a custom secret volume
+	if spec.TLS != nil && spec.TLS.SecretName != "" {
```

Comment:
I'm confused what would happen here, if one uses your manifest [example](https://github.com/zalando/postgres-operator/blob/d152ea9a4a57e853c7ac20f55eb27189fb547b09/manifests/complete-postgres-manifest.yaml#L106) where the secret name is empty. Because, I've followed your documentation and created a secret, but then only used the complete manifest to see that nothing happens to the stateful set

Reply:
I think I assumed that file was used to display the defaults. In my mind it wasn't going to be used as an example to setup TLS. Should I add a comment next to the empty value to say that it's disabled?

Reply:
It's [this](https://github.com/zalando/postgres-operator/blob/d152ea9a4a57e853c7ac20f55eb27189fb547b09/manifests/complete-postgres-manifest.yaml#L104) comment, that confuses me. To me it sounds like: "If you leave it blank, the TLS secret will be generated and used by the pods"

Similar to e.g. service accounts in our helm [chart](https://github.com/zalando/postgres-operator/blob/51909204fd7085f5b890cc4689221fece54ccecc/charts/postgres-operator/values.yaml#L261).

So yeah adding a comment behind the field in the manifest sounds good + in the [reference docs](https://github.com/zalando/postgres-operator/blob/master/docs/reference/cluster_manifest.md). Btw, just noticed that the reference does not cover the TLS section at all. Please add a short explanation for all fields.

Reply:
Ok I see :+1: . Fix incoming.

Reply:
Ok I added some extra documentation, let me know what you think.

Reply:
Thank you :)
Just some minor comments and I think it's done then